pkg/ceph/osd/daemon.go

```diff
@@ -38,6 +38,11 @@ const (
 )
 
 func Run(context *clusterd.Context, agent *OsdAgent) error {
+	if err := setHostName(context); err != nil {
```

Comment:
This is a re-implementation of #598 to support the tini change. We need a consistent hostname for the osd pod to put in the crush map